### PR TITLE
bench: indirect call benchmark

### PR DESCRIPTION
**Pull request by axic**
It gets inefficiently compiled to exactly what we want to test:
```webassembly
  (func $fnv1 (type $t0) (param $p0 i64) (param $p1 i64) (result i64)
    get_local $p1
    get_local $p0
    i64.xor
    i64.const 1099511628211
    i64.mul)
  (func $threeab (type $t0) (param $p0 i64) (param $p1 i64) (result i64)
    get_local $p0
    i64.const 3
    i64.mul
    get_local $p1
    i64.add)
  (func $icall (export "icall") (type $t2) (param $p0 i32) (result i32)
    (local $l0 i64) (local $l1 i32)
    block $B0
      get_local $p0
      i32.eqz
      br_if $B0
      i64.const -3750763034362895579
      set_local $l0
      i32.const 0
      set_local $l1
      loop $L1
        get_local $l0
        i64.const 1311768467294899695
        get_local $l1
        i32.const 1
        i32.and
        i32.const 2
        i32.shl
        i32.const 1024
        i32.add
        i32.load
        call_indirect (type $t0)
        set_local $l0
        get_local $p0
        get_local $l1
        i32.const 1
        i32.add
        tee_local $l1
        i32.ne
        br_if $L1
      end
      get_local $l0
      i32.wrap/i64
      return
    end
    i32.const -2078137563)
```

### CODE REVIEW

**Comment by axic** [comment]
Quick run:
master:
```
fizzy/execute/micro/icall_hash/1000_steps         335 us          330 us         2255
 wabt/execute/micro/icall_hash/1000_steps         356 us          353 us         2015
wasm3/execute/micro/icall_hash/1000_steps        37.9 us         37.6 us        18619
```

#616:
```
fizzy/execute/micro/icall_hash/1000_steps         131 us          121 us         5731
 wabt/execute/micro/icall_hash/1000_steps         342 us          340 us         2080
wasm3/execute/micro/icall_hash/1000_steps        38.2 us         37.6 us        19328
```

File: test/benchmarks/micro/icall_hash.c [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+}
+
+uint64_t threeab(uint64_t state, uint64_t input)
+{
+    return (3 * state) + input;
+}
+
+typedef uint64_t (*hashfn)(uint64_t state, uint64_t input) ;
+
+static hashfn fns[] = { fnv1, threeab };
```
Comment (axic) [inline test/benchmarks/micro/icall_hash.c:17]: Could also just use `{ fnv1, fnv1 };` if we want to reduce the surface area.

File: test/benchmarks/micro/icall_hash.c [hunk 2]
```diff
@@ -0,0 +1,27 @@
+#include <inttypes.h>
```
Comment (chfast) [inline test/benchmarks/micro/icall_hash.c:1]:
```suggestion
#include <stdint.h>
```

File: test/benchmarks/micro/icall_hash.c (hunk 1 above)
Comment (chfast) [reply test/benchmarks/micro/icall_hash.c:17]: This is fine.

Did you format the code?

Comment (axic) [reply test/benchmarks/micro/icall_hash.c:17]: With clang-format? No, will do.

**Review by chfast — APPROVED** [review]

**Review by gumb0 — APPROVED** [review]

**Pull request merged**